connector/src/main/java/org/geysermc/connector/network/session/GeyserSession.java

```diff
@@ -751,4 +803,109 @@ public void sendAdventureSettings() {
         adventureSettingsPacket.getSettings().addAll(flags);
         sendUpstreamPacket(adventureSettingsPacket);
     }
+
+    /**
+     * Updates the stored bounding box
+     * @param position The new position of the player
+     */
+    public void updatePlayerBoundingBox(Vector3f position) {
+        updatePlayerBoundingBox(Vector3d.from(position.getX(), position.getY(), position.getZ()));
+    }
+
+    /**
+     * Updates the stored bounding box
+     * @param position The new position of the player
+     */
+    public void updatePlayerBoundingBox(Vector3d position) {
+        updatePlayerBoundingBox();
+
+        playerBoundingBox.setMiddleX(position.getX());
+        playerBoundingBox.setMiddleY(position.getY() + (playerBoundingBox.getSizeY() / 2));
+        playerBoundingBox.setMiddleZ(position.getZ());
+    }
+
+    /**
+     * Updates the stored bounding box without passing a position, which currently just changes the height depending on if the player is sneaking.
+     */
+    public void updatePlayerBoundingBox() {
+        if (playerBoundingBox == null) {
+            Vector3f playerPosition;
+            if (playerEntity == null) {
+                // Temporary position to prevent NullPointerException
+                playerPosition = Vector3f.ZERO;
+            } else {
+                playerPosition = playerEntity.getPosition();
+            }
+            playerBoundingBox = new BoundingBox(playerPosition.getX(), playerPosition.getY() + 0.9, playerPosition.getZ(), 0.6, 1.8, 0.6);
+        } else {
+            // According to the Minecraft Wiki, when sneaking:
+            // - In Bedrock Edition, the height becomes 1.65 blocks, allowing movement through spaces as small as 1.75 (2 - 1⁄4) blocks high.
+            // - In Java Edition, the height becomes 1.5 blocks.
+            if (sneaking) {
+                playerBoundingBox.setSizeY(1.5);
+            } else {
+                playerBoundingBox.setSizeY(1.8);
+            }
+        }
+    }
+
+    /**
+     * Additional space where blocks are checked, which is helpful for fixing NoCheatPlus's Passable check.
+     * This check doesn't allow players right up against the block, so they must be pushed slightly away.
+     */
+    public static final double COLLISION_TOLERANCE = 0.00001; // TODO: Move?
```

Comment:
Static variables should be at the top of the file.

Reply:
Done